alchemiscale_fah/compute/client.py

```diff
@@ -0,0 +1,589 @@
+
+    def create_project(self, project_id, project_data: ProjectData):
+        self._put(self.ws_url, f"/api/projects/{project_id}", project_data.dict())
+
+    def update_project(self, project_id, project_data: ProjectData):
+        self.create_project(project_id, project_data)
+
+    def delete_project(self, project_id):
+        self._delete(self.ws_url, f"/api/projects/{project_id}")
+
+    def get_project(self, project_id) -> ProjectData:
+        return ProjectData(**self._get(self.ws_url, f"/api/projects/{project_id}"))
+
+    def list_project_files(self, project_id) -> list[FileData]:
+        """Get a list of files associated with a project.
+
+        Parameters
+        ----------
+        project_id
+            ID of the project
+        src
+            File to download.
+        dest
+            Path to download file to.
+
+        """
+        return [
+            FileData(**i)
+            for i in self._get(
+                self.ws_url,
+                f"/api/projects/{project_id}/files",
+            )
+        ]
+
+    def create_project_file(self, project_id, src: Path, dest: Path):
+        """Upload a file to the PROJECT directory tree.
+
+        Parameters
+        ----------
+        project_id
+            ID of the project
+        src
+            File to upload.
+        dest
+            Path relative to PROJECT directory to upload to.
+
+        """
+        self._upload(
+            self.ws_url,
+            f"/api/projects/{project_id}/files/{dest}",
+            src,
+        )
+
+    def create_project_file_from_bytes(self, project_id, bytedata: bytes, dest: Path):
+        self._upload_bytes(
+            self.ws_url,
+            f"/api/projects/{project_id}/files/{dest}",
+            bytedata,
+        )
+
+    def delete_project_file(self, project_id, path):
+        """Delete a file from the PROJECT directory tree.
+
+        Parameters
+        ----------
+        project_id
+            ID of the project
+        path
+            File to delete.
+
+        """
+        self._delete(
+            self.ws_url,
+            f"/api/projects/{project_id}/files/{path}",
+        )
+
+    def get_project_file(self, project_id, src, dest):
+        """Download a file from the PROJECT directory tree.
+
+        Parameters
+        ----------
+        project_id
+            ID of the project
+        src
+            File to download.
+        dest
+            Path to download file to.
+
+        """
+        self._download(self.ws_url, f"/api/projects/{project_id}/files/{src}", dest)
+
+    def get_project_file_to_bytes(self, project_id, src):
+        return self._download_bytes(
+            self.ws_url, f"/api/projects/{project_id}/files/{src}"
+        )
+
+    def get_project_jobs(self, project_id, since: datetime = None) -> JobResults:
+        """Get all active jobs for the project.
+
+        Parameters
+        ----------
+        project_id
+            ID of the project
+        since
```

Comment:
Should be `Optional[datetime]`, no specific format needed since this is a `datetime.datetime` object. I've updated the type hint to include `Optional[...]` since this was missing!